### PR TITLE
Add waveform quality check.

### PR DESCRIPTION
**Pull request by marcomuzio**
Added a module to perform quality checks on the waveform itself. To avoid errors during processing this needs to be separate from the daq quality check. Currently this just checks for waveform saturation, but wrote it so other checks can be easily added later.

### CODE REVIEW

**Review by drparamita — COMMENTED** [review]
Looks fine.

File: araproc/analysis/waveform_quality_cut.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    -------
+    is_saturated : bool
+        Whether a waveform is saturated.
+    """
+    
+    chans = list(wave_bundle.keys())
+    
+    # set parameters for saturation check
+    n_points = 5
+    saturation_threshold = 1500 # mV
```
Comment (clark2668) [inline araproc/analysis/waveform_quality_cut.py:28]: Did you confirm that this fires? I wonder if we want to be a little more generous, like 1495 or something?

Comment (clark2668) [reply araproc/analysis/waveform_quality_cut.py:28]: Okay, great, I'm glad it grabbed the sample. Event. Can you add a comment to the code about which one? And okay, fair enough re: 1500

Comment (marcomuzio) [reply araproc/analysis/waveform_quality_cut.py:28]: So you want something like 
`# Example saturation event: A4 run 6421 event 15333`?

Comment (clark2668) [reply araproc/analysis/waveform_quality_cut.py:28]: Exactly

Comment (marcomuzio) [reply araproc/analysis/waveform_quality_cut.py:28]: Okay done in [https://github.com/ara-software/AraProc/pull/71](PR#71)

Comment (clark2668) [reply araproc/analysis/waveform_quality_cut.py:28]: Great, thank you.

**Pull request merged**